Source/WebCore/html/HTMLSelectElement.cpp

```diff
@@ -1658,4 +1659,33 @@ unsigned HTMLSelectElement::length() const
     return options;
 }
 
+ExceptionOr<void> HTMLSelectElement::showPicker()
+{
+    auto* frame = document().frame();
+    if (!frame)
+        return { };
+
+    if (!isMutable())
+        return Exception { InvalidStateError, "Select showPicker() cannot be used on immutable controls."_s };
+
+    // In cross-origin iframes it should throw a "SecurityError" DOMException. In same-origin iframes it should work fine.
+    auto* localTopFrame = dynamicDowncast<LocalFrame>(frame->tree().top());
+    if (!localTopFrame || !frame->document()->securityOrigin().isSameOriginAs(localTopFrame->document()->securityOrigin()))
+        return Exception { SecurityError, "Select showPicker() called from cross-origin iframe."_s };
+
+    auto* window = frame->window();
+    if (!window || !window->hasTransientActivation())
+        return Exception { NotAllowedError, "Select showPicker() requires a user gesture."_s };
+
+#if PLATFORM(IOS_FAMILY)
+    focus();
```

Comment:
The showPicker spec isn't clear on if this is allowed or not (see https://github.com/whatwg/html/issues/9757) but given the spirit of the feature it feels like working rather than not working is the best solution.

Reply:
Moving this into a follow up that addresses showPicker on iOS in general